base/essentials.jl

```diff
@@ -539,6 +539,17 @@ julia> sizeof(1.0)
 
 julia> sizeof(collect(1.0:10.0))
 80
+
+julia> struct StructWithPadding
+           x::Int64
+           flag::Bool
+       end
+
+julia> sizeof(StructWithPadding) # not the sum of `sizeof` of fields due to padding
+16
```

Comment:
Keno had proposed changing this (https://github.com/JuliaLang/julia/pull/46322), though we didn't finish

Reply:
The exact number here is definitely an implementation detail. For example, on 32-bit platforms this returns:
```
julia> sizeof(StructWithPadding)
12
```